eclair-core/src/main/scala/fr/acinq/eclair/db/PaymentsDb.scala

```diff
@@ -119,22 +127,26 @@ object IncomingPaymentStatus {
  * An outgoing payment sent by this node.
  * At first it is in a pending state, then will become either a success or a failure.
  *
- * @param id             internal payment identifier.
- * @param parentId       internal identifier of a parent payment, or [[id]] if single-part payment.
- * @param externalId     external payment identifier: lets lightning applications reconcile payments with their own db.
- * @param paymentHash    payment_hash.
- * @param amount         amount of the payment, in milli-satoshis.
- * @param targetNodeId   node ID of the payment recipient.
- * @param createdAt      absolute time in milli-seconds since UNIX epoch when the payment was created.
- * @param paymentRequest Bolt 11 payment request (if paying from an invoice).
- * @param status         current status of the payment.
+ * @param id              internal payment identifier.
+ * @param parentId        internal identifier of a parent payment, or [[id]] if single-part payment.
+ * @param externalId      external payment identifier: lets lightning applications reconcile payments with their own db.
+ * @param paymentHash     payment_hash.
+ * @param paymentType     distinguish different payment types (standard, swaps, etc).
+ * @param amount          amount that will be received by the target node.
+ * @param finalAmount     amount that will be received by the final recipient.
+ * @param recipientNodeId id of the final recipient.
+ * @param createdAt       absolute time in milli-seconds since UNIX epoch when the payment was created.
+ * @param paymentRequest  Bolt 11 payment request (if paying from an invoice).
+ * @param status          current status of the payment.
  */
 case class OutgoingPayment(id: UUID,
                            parentId: UUID,
                            externalId: Option[String],
                            paymentHash: ByteVector32,
+                           paymentType: Option[String],
```

Comment:
This needs to be made mandatory imo, which may require a migration but makes future but sets the record clean.

Reply:
Fixed in `3b6f468`